RecursionError: maximum recursion depth exceeded in python 3.6

Hi,
I am using django-bitfield 1.9.3 in a Django 1.10.7 app in conjunction with django-filter 1.0.4 (and a lot others). I am porting the code from python 2.7 to python 3.6; actually, for the moment, make it work on both python versions.

Now, we have a BitField and a ChoiceFilter filter, something like:
```
test = BitField(flags=['a', 'b', 'c'], db_index=True)

and

class BitFlagsFilter(ChoiceFilter):
    def __init__(self, *args, **kwargs):
        self.flags = kwargs.pop('flags', None)
        self.generated_choices = tuple(enumerate(self.flags.keys())) if self.flags else ()
        super(BitFlagsFilter, self).__init__(*args, choices=self.generated_choices, **kwargs)
```

On 2.7 works like a charm, on 3.6 it breaks, from within django filters:
```
/python3.6/site-packages/django_filters/filterset.py in __init__
186. self.filters = copy.deepcopy(self.base_filters) 
...
/python3.6/site-packages/bitfield/models.py in __getattr__
31. if key not in self._flags: ...
Error in formatting: RecursionError: maximum recursion depth exceeded
```


I know that it breaks from filters, but please take a look at this debuging:

- python 2.7
```
(Pdb) self.flags
['a', 'b', 'c']
(Pdb) type(self.flags)
<class 'bitfield.models.BitFieldFlags'>
(Pdb) from copy import deepcopy
(Pdb) deepcopy(self.flags)
['a', 'b', 'c']
```

- python 3.6
```
(Pdb) self.flags
['a', 'b', 'c']
(Pdb) type(self.flags)
<class 'bitfield.models.BitFieldFlags'>
(Pdb) from copy import deepcopy
(Pdb) deepcopy(self.flags)
*** RecursionError: maximum recursion depth exceeded
```

Any thoughts about this? Any help/suggestion will be highly appreciated.

A short update:

Even though two test are failing on python 3.6
```
======================================================================
FAIL: test_negate (bitfield.tests.tests.BitFieldTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "../python3.6/site-packages/bitfield/tests/tests.py", line 215, in test_negate
    self.assertEqual(BitFieldTestModel.objects.filter(flags=~BitFieldTestModel.flags.FLAG_0).count(), 1)
AssertionError: 0 != 1

======================================================================
FAIL: test_update (bitfield.tests.tests.BitFieldTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "../python3.6/site-packages/bitfield/tests/tests.py", line 187, in test_update
    self.assertTrue(instance.flags.FLAG_1)
AssertionError: <Bit: number=1, is_set=False> is not true
```
I found a fix for the reported error which actually works for me:
```
class BitFieldFlags(object):
...
    def __deepcopy__(self, memo):
        cls = self.__class__
        result = cls.__new__(cls)
        memo[id(self)] = result
        for k, v in self.__dict__.items():
            setattr(result, k, deepcopy(v, memo))
        return result
```

--- a/bitfield/models.py
+++ b/bitfield/models.py
@@ -28,9 +28,24 @@ class BitFieldFlags(object):
             yield flag
 
     def __getattr__(self, key):
-        if key not in self._flags:
+        if key == '_flags':
+            # Since __getattr__ is for fallback, reaching here from Python
+            # means that there's no '_flags' attribute in this object,
+            # which may be caused by intermediate state while copying etc.
+            raise AttributeError(
+                "'%s' object has no attribute '%s'" % (self.__class__.__name__, key)
+            )
+        try:
+            flags = self._flags
+        except AttributeError:
+            raise AttributeError(
+                "'%s' object has no attribute '%s'" % (self.__class__.__name__, key)
+            )
+        try:
+            flag = flags.index(key)
+        except ValueError:
             raise AttributeError("flag {} is not registered".format(key))
-        return Bit(self._flags.index(key))
+        return Bit(flag)
 
     def iteritems(self):
         for flag in self._flags:

